forge-game/src/main/java/forge/game/trigger/TriggerExiled.java

```diff
@@ -77,6 +79,43 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
             return false;
         }
 
+        if (hasParam("WhileKeyword")) {
+            final String keyword = getParam("WhileKeyword");
+            boolean withKeyword = false;
+
+            IndividualCostPaymentInstance currentPayment = (IndividualCostPaymentInstance) runParams.get(AbilityKey.IndividualCostPaymentInstance);
+
+            SpellAbility sa;
+            if (currentPayment != null) {
+                sa = currentPayment.getPayment().getAbility();
+
+                if (sa != null && sa.getHostCard() != null) {
+                    if ((sa.isSpell() || sa.isAbility()) && sa.getHostCard().hasStartOfUnHiddenKeyword(keyword)) {
```

Comment:
just added `|| sa.isAbility()` for `WhileKeyword$ Craft` to work...

is the logic lines 99-113 needed in this case?

Reply:
> Most of the time, Market Gnome's last ability will trigger because it was exiled as one of the components used in the craft ability. However, in some uncommon cases, the ability will also trigger if Market Gnome is exiled while you're activating mana abilities as part of paying for that craft ability.

maybe the check should be even stricter and use `isCraft()`

Reply:
I think I've got it wired up right, but I can't dream up a way to test this 😅

Reply:
<img width="1088" alt="image" src="https://github.com/Card-Forge/forge/assets/103371817/1d811f95-f46a-4dfc-b9fe-6bfb6e7c3f8a">
Market Gnome used to help pay mana cost!